lib/src/components/NxProgressBar/__tests__/NxProgressBar.test.tsx

```diff
@@ -4,191 +4,176 @@
     });
 
-    it('adds the appropriate modifier class to each step depending on whether it is above, below, or at the ' +
-         'current value', function() {
-      const zeroBar = getMountedComponent({ showSteps: true, max: 12, value: 0 }),
-          partialBar = getMountedComponent({ showSteps: true, max: 12, value: 3 }),
-          fullBar = getMountedComponent({ showSteps: true, max: 12, value: 12 });
+    it('is overriden by labelError when it is specified', function() {
+      const el = renderEl()!,
+          elWithError = renderEl({ labelError: 'oops!' })!;
 
-      expect(zeroBar.find('.nx-progress-bar__step--below-value')).not.toExist();
-      expect(zeroBar.find('.nx-progress-bar__step--at-value')).not.toExist();
-      expect(zeroBar.find('.nx-progress-bar__step--above-value').length).toBe(11);
+      expect(el).toHaveTextContent('current progress');
+      expect(elWithError).not.toHaveTextContent('current progress');
+      expect(elWithError).toHaveTextContent('oops!');
+    });
+  });
+
+  it('sets progress to 0 when labelError is defined', function() {
+    const elWithError = renderEl({ labelError: 'oops!', value: 50 })!;
+    expect(within(elWithError).getByRole('progressbar')).toHaveAttribute('value', '0');
+    expect(elWithError).toHaveTextContent('0%');
+  });
 
-      expect(partialBar.find('.nx-progress-bar__step--below-value').length).toBe(2);
-      expect(partialBar.find('.nx-progress-bar__step--at-value').length).toBe(1);
-      expect(partialBar.find('.nx-progress-bar__step--above-value').length).toBe(8);
+  it('assigns an aria-label when component\'s label text is not rendered', function() {
+    const progressBarWithLabel = quickRender().getByRole('progressbar');
+    const progressBarWithoutLabel = quickRender({ inlineCounter: true }).getByRole('progressbar');
 
-      // ensure correct ordering
-      expect(partialBar.find('.nx-progress-bar__step--at-value ~ .nx-progress-bar__step--below-value')).not.toExist();
-      expect(partialBar.find('.nx-progress-bar__step--above-value ~ .nx-progress-bar__step--at-value')).not.toExist();
+    expect(progressBarWithLabel).not.toHaveAttribute('aria-label', 'current progress');
+    expect(progressBarWithoutLabel).not.toHaveTextContent('current progress');
+    expect(progressBarWithoutLabel).toHaveAttribute('aria-label', 'current progress');
+  });
+
+  describe('showSteps', function() {
+    it('renders an additional child div with the presentation role', function() {
+      const elWithoutSteps = renderEl()!,
+          elWithSteps = renderEl({ showSteps: true })!;
+
+      expect(elWithoutSteps.childElementCount).toBe(2);
+      expect(within(elWithoutSteps).queryByRole('presenation')).not.toBeInTheDocument();
+
+      expect(elWithSteps.childElementCount).toBe(3);
+      expect(within(elWithSteps).getByRole('presentation')).toBeInTheDocument();
+    });
+
+    it('doesn\'t render a counter when showSteps is true', function() {
```

Comment:
Of the three tests in this describe block, this is the only one that doesn't seem strictly stylistic to me.  I'd delete the other two and double check that they're covered by visual tests

Reply:
> Of the three tests in this describe block, this is the only one that doesn't seem strictly stylistic to me. I'd delete the other two and double check that they're covered by visual tests

The third test is shown in visual tests (number of steps with assigned max value in label both visible). The role in the first test can't be tested visually, but if you don't think it's necessary in testing I can remove that test as well.

Reply:
> The third test is shown in visual tests (number of steps with assigned max value in label both visible). The role in the first test can't be tested visually, but if you don't think it's necessary in testing I can remove that test as well.

`presentation` is the role for things that aren't relevant to accessibility technologies.  In fact `none` is an exact synonym for it in the role names.  So pretty much by definition anything with the `presentation` role is something we don't need to test in the unit tests.

Reply:
> `presentation` is the role for things that aren't relevant to accessibility technologies. In fact `none` is an exact synonym for it in the role names. So pretty much by definition anything with the `presentation` role is something we don't need to test in the unit tests.

Thanks for clarifying there. So I've removed the two tests ([7258a9d](https://github.com/sonatype/sonatype-react-shared-components/pull/903/commits/7258a9d0e446193811bd5c98ce8978372037026e)) , and removed the `describe` block since it would just be holding one test. I did however create a new `describe` block for the counter to group the 4 tests together.